misc/python/materialize/checks/all_checks/continual_task.py

```diff
@@ -0,0 +1,181 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+from textwrap import dedent
+
+from materialize.checks.actions import Testdrive
+from materialize.checks.checks import Check
+from materialize.checks.common import KAFKA_SCHEMA_WITH_SINGLE_STRING_FIELD
+from materialize.checks.executors import Executor
+from materialize.mz_version import MzVersion
+
+
+def schemas() -> str:
+    return dedent(KAFKA_SCHEMA_WITH_SINGLE_STRING_FIELD)
+
+
+class AuditLogCT(Check):
+    """Continual Task for audit logging"""
+
+    def _can_run(self, e: Executor) -> bool:
+        return self.base_version >= MzVersion.parse_mz("v0.120.0-dev")
+
+    def initialize(self) -> Testdrive:
+        return Testdrive(
+            schemas()
+            + dedent(
+                """
+                > CREATE TABLE t_input (key INT);
+                > INSERT INTO t_input VALUES (1);
+                > CREATE MATERIALIZED VIEW anomalies AS SELECT sum(key)::INT FROM t_input;
+                > CREATE CONTINUAL TASK audit_log (count INT) ON INPUT anomalies AS (
+                    INSERT INTO audit_log SELECT * FROM anomalies WHERE sum > 1;
+                  )
+                """
+            )
+        )
+
+    def manipulate(self) -> list[Testdrive]:
+        return [
+            Testdrive(schemas() + dedent(s))
+            for s in [
+                """
+                > INSERT INTO t_input VALUES (2), (3);
+                """,
+                """
+                > INSERT INTO t_input VALUES (4), (5), (6);
+                """,
+            ]
+        ]
+
+    def validate(self) -> Testdrive:
+        return Testdrive(
+            dedent(
+                """
+                > SELECT * FROM audit_log
+                1
+                6
+                21
+           """
+            )
+        )
+
+
+class StreamTableJoinCT(Check):
+    """Continual Task for stream table join"""
+
+    def _can_run(self, e: Executor) -> bool:
+        return self.base_version >= MzVersion.parse_mz("v0.118.0-dev")
+
+    def initialize(self) -> Testdrive:
+        return Testdrive(
+            schemas()
+            + dedent(
+                """
+                > CREATE TABLE big (key INT);
+                > CREATE TABLE small (key INT, val STRING);
+                > INSERT INTO small VALUES (1, 'v1');
```

Comment:
Having data in `small` before we create the CT is fine because small is being used as a "reference" not an "input" (sorry for the jargon) and the non-empty thing only matters for stuff listen in `ON INPUT`